cadc-vos-server/src/test/java/org/opencadc/vospace/server/VospacePackageRunnerTest.java

```diff
@@ -0,0 +1,130 @@
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.vospace.server;
+
+import ca.nrc.cadc.util.Log4jInit;
+import java.net.URI;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opencadc.vospace.server.pkg.VospacePackageRunner;
+
+public class VospacePackageRunnerTest {
+
+    private static final Logger log = Logger.getLogger(org.opencadc.vospace.server.UtilsTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.raven", Level.DEBUG);
+    }
+
+    @Test
+    public void testIsTargetInPackage() throws Exception {
+
+        // should fail
+        URI node = URI.create("vos://opencadc.org~vault/pkg/foo/link");
+        URI target = URI.create("http://example.com/target");
+        String relativePath = "/pkg/foo/";
+        boolean result = VospacePackageRunner.isTargetInPackage(node, target, relativePath);
+        Assert.assertFalse(result);
+
+        node = URI.create("vos://opencadc.org~vault/pkg/foo/link");
+        target = URI.create("vos://example.com~vault/pkg/target");
```

Comment:
this target is outside for two reasons so doesn't necessarily test both

Reply:
Not seeing the second reason.
In the context where isTargetInPackage is called, the relative path is the relative path to the link node parent. The relative path and the link node name are used to compose the relative path to the node, which is passed to the PackageItem constructor. 
Although it might be better if it was the relative path to the link node in isTargetInPackage, and not the parent. Might change that.

Reply:
outside the package (path) or outside the service (external link)

I am interpreting the test as `node` is the root of the package and `target` is the target of the link. Maybe part of the issue is that `node` might as well be `n` or `obj` because it doesn't convey any meaning at all... and it's not even a Node, it's a URI. Which node is it?

1. the target has a different authority
2. the child path is not under the node path

iirc, a different test checks "different authority" ... well I thought so, but all the node and target have different authority so now I really don't know what's being tested here